pkg/net/libp2p/key.go

```diff
@@ -27,15 +27,32 @@ func operatorPrivateKeyToNetworkKeyPair(operatorPrivateKey *operator.PrivateKey)
 		return nil, nil, fmt.Errorf("libp2p supports only secp256k1 operator keys")
 	}
 
-	// Libp2p keys are actually btcec keys under the hood.
-	btcecPrivateKey, btcecPublicKey := btcec.PrivKeyFromBytes(
-		DefaultCurve, operatorPrivateKey.D.Bytes(),
+	// Note that `libp2pcrypto.UnmarshalSecp256k1PrivateKey` uses the secp256k1
+	// implementation provided by decred underneath
+	privKey, err := libp2pcrypto.UnmarshalSecp256k1PrivateKey(
+		operatorPrivateKey.D.Bytes(),
 	)
+	if err != nil {
+		return nil, nil, err
```

Comment:
Let's wrap the error with some additional context. This will be helpful during debugging:
```suggestion
		return nil, nil, fmt.Errorf("cannot unmarshal operator private key as a libp2p secp256k1 private key: [%v]", err)
```

Reply:
Done in aebfadd057602ab9dcf5e0fc59e4e95b890d750a.